spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/EnvironmentEntryInitializingTreeMap.java

```diff
@@ -90,4 +90,8 @@ public T put(String key, T value) {
 		return delegate.entrySet();
 	}
 
+	@Override
+	public boolean containsKey(Object key) {
```

Comment:
This effectively returns true every time. I wonder if we shouldn't rather do a 'get' inside `BindingServiceProperties`.